src/main/java/org/jabref/logic/importer/fetcher/Medra.java

```diff
@@ -0,0 +1,173 @@
+                                                                    .withField(StandardField.YEAR,
+                                                                               Optional.ofNullable(item.optJSONObject("issued"))
+                                                                                       .map(array -> array.optJSONArray("date-parts"))
+                                                                                       .map(array -> array.optJSONArray(0))
+                                                                                       .map(array -> array.optInt(0))
+                                                                                       .map(year -> Integer.toString(year)).orElse(""))
+                                                                    .withField(StandardField.DOI, item.getString("DOI"))
+                                                                    .withField(StandardField.PAGES, item.optString("page"))
+                                                                    .withField(StandardField.ISSN, item.optString("ISSN"))
+                                                                    .withField(StandardField.JOURNAL, item.optString("container-title"))
+                                                                    .withField(StandardField.PUBLISHER, item.optString("publisher"))
+                                                                    .withField(StandardField.URL, item.optString("URL"))
+                                                                    .withField(StandardField.VOLUME, item.optString("volume"));
+
+        } catch (JSONException exception) {
+            throw new ParseException("mEdRA API JSON format has changed", exception);
+        }
+    }
+
+    private EntryType convertType(String type) {
+        switch (type) {
+            case "article-journal":
+                return StandardEntryType.Article;
+            default:
+                return StandardEntryType.Misc;
+        }
+    }
+
+    private String toAuthors(JSONArray authors) {
+        if (authors == null) {
+            return "";
+        }
+
+        // input: list of {"literal":"A."}
+        AuthorList authorsParsed = new AuthorList();
+        String name = "";
+
+        for (int i = 0; i < authors.length(); i++) {
+            JSONObject author = authors.getJSONObject(i);
+            name = author.optString("literal", "") + " " + author.optString("family", "") + " " + author.optString("given", "");
+
+            authorsParsed.addAuthor(
+                                    name,
+                                    "",
+                                    "",
+                                    "",
+                                    "");
+
+        }
+        return authorsParsed.getAsFirstLastNamesWithAnd();
+    }
+
+    @Override
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
```

Comment:
Same question here as above; normally you don't need to overwrite the `performSearchById` method.

Reply:
In this case I have overridden the method because I needed to add a header attribute to URLDownload and IdBasedParserFetcher does not have a getUrlDownload to override. It invokes URLDownload constructor directly inside the try resource, while for example SearchBasedParserFetcher invokes getUrlDownload method.

Reply:
Fixed in https://github.com/mind000/jabref/pull/1.

Reply:
The `getUrlDownload` method in `SearchBasedParserFetcher` was added exactly for this reason. As this seems to be a common issue for fetchers, the `IdBasedParserFetcher` should be changed accordingly. Then there shouldn't be any need to overwrite the `performSearchById` method, but only `getUrlDownload`.
https://github.com/JabRef/jabref/blob/d6244d71561a4515697f198468fd671c90d922e2/src/main/java/org/jabref/logic/importer/SearchBasedParserFetcher.java#L52-L62

Reply:
@tobiasdiez Fixed in https://github.com/mind000/jabref/pull/1. Due to multi-inheritcance, I had to name the method `getUrlDownloadForIdentifier` (in this way, it is distinguishable from `getUrlDownloadForQuery`. Both accept `String` parameter and return a `URLDownload). deep link to the commit https://github.com/mind000/jabref/pull/1/commits/6fd81c3fd4f36610835097f39c91ceb77769bb05